### PR TITLE
Simple models + views

### LINKED ISSUE #39: Create Officer model and CRUD controller
## Model Name
Officer

### Columns
- [x]  fist_name
- [x]  last_name
- [x] DoB
- [x] Badge_Number
- [x] Nationality
- [x] race
- [x] Gender
- [x] Force
- [x]  Active (boolean)

**Describe CRUD Setup**
- [ ] ADMINS only should be able to add, update and delete officers, however all users should be able to see officers

**Additional Endpoints Required**
- [ ]  _Good Eggs_: show active officers sorted with the highest number number of positive incidents shown first
- [ ] _Bad Apples_: show active officers with the highest number of incidents negative shown first

**Add Model Link to Landing Page**
- [ ] Please add a link to `/officers` in: [Landing Page](https://github.com/My-Power-STN39M-Hopefully/django-good-egg-backend/blob/development/good_egg/templates/landing_page.html)

_Outside Help_
[Django Model Documentation](https://docs.djangoproject.com/en/2.1/ref/models/fields/)
[GA Django REST Framework Tutorial](https://git.generalassemb.ly/seir-622/django-rest-framework)

### LINKED ISSUE #40: Create Incident model and CRUD controller
## Model Name
Incident

### Columns
- [x]  Description (required) 
- [x]  Category (required)
- [x]  Officers
- [ ]  Officer description (required IF officers null)
- [x]  Date (required)
- [x]  Time
- [x]  Location
- [x]  Formal_Complaint(boolean)
- [x]  Formal_Complaint_Number
- [x]  Witnesses_Present(boolean)
- [x]  Witnesses_Information
- [x]  User_id (required)
- [x]  Private (boolean, if true only admins and incident owners will see)
- [x]  bad_apple _True false value, defaults to true, if false incident will count as a "good egg" incident and add to an officers "good egg" rating_

**Describe CRUD Setup**
- [ ] For non-admins they should only be able to update their own incidents and if a incident is private (defaults to false) then only the owner of the incident should be able to see it.

**Additional Endpoints Required**
- [ ] There should be an endpoint that returns the incidents created in the last two weeks, make sure that the user has access to that incident.

**Add Model Link to Landing Page**
- [ ] Please add a link to `/incidents` in: [Landing Page](https://github.com/My-Power-STN39M-Hopefully/django-good-egg-backend/blob/development/good_egg/templates/landing_page.html)

_Outside Help_
[Django Model Documentation](https://docs.djangoproject.com/en/2.1/ref/models/fields/)
[GA Django REST Framework Tutorial](https://git.generalassemb.ly/seir-622/django-rest-framework)

### PR DESCRIPTION
**Pull request by MorganMac1993**
closes #39 closes #40

### CODE REVIEW

File: good_egg/models.py [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+    (YES, 'yes'),
+    (NO, 'no'),
+  )
+  category = models.CharField(max_length=100)
+  officers =  models.ForeignKey(Officer, on_delete=models.CASCADE, related_name='officer')
+  date = models.CharField(max_length=100)
+  time = models.CharField(max_length=100)
+  location = models.CharField(max_length=100)
+  formal_complaint = models.CharField(max_length=100)
+  formal_complaint_number = models.CharField(max_length=100, choices=yes_or_no, default=NO)
```
Comment (Cassioblu55) [inline good_egg/models.py:17]: Should this have `choices=yes_or_no`?

File: good_egg/models.py [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+  witnesses_present = models.CharField(max_length=100, choices=yes_or_no, default=NO)
+  witnesses_information = models.CharField(max_length=100)
+  User_id = models.CharField(max_length=100)
+  private = models.CharField(max_length=100, choices=yes_or_no, default=NO)
```
Comment (Cassioblu55) [inline good_egg/models.py:21]: Not saying you should change but why not just use `BooleanField `?

File: good_egg/views.py [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+from django.shortcuts import render, redirect
+from .models import Incident
+from .models import Officer
+from django.contrib.auth.decorators import login_required
 
 # Create your views here.
 def landing_page(request):
     return render(request, 'landing_page.html')
+
+def incident_list(req):
```
Comment (Cassioblu55) [inline good_egg/views.py:10]: Please use [Django rest framework views](https://git.generalassemb.ly/seir-622/django-rest-framework#views-20-min--140)

**Review by Cassioblu55 — CHANGES_REQUESTED** [review]
Please use [Django Rest Framework API Views](https://git.generalassemb.ly/seir-622/django-rest-framework#views-20-min--140)

File: good_egg/views.py (hunk 3 above)
Comment (MorganMac1993) [reply good_egg/views.py:10]: absolutely

File: good_egg/models.py (hunk 2 above)
Comment (MorganMac1993) [reply good_egg/models.py:21]: I think I was just on a role.  I can do boolean field instead forsure.

File: good_egg/models.py (hunk 1 above)
Comment (MorganMac1993) [reply good_egg/models.py:17]: No i guess not.  I think default=N/A works just as well

**Pull request closed without merging**